src/Services/CleanupService.php

```diff
@@ -0,0 +1,390 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: kennyva
+ * Date: 10/10/2018
+ * Time: 12:47
+ */
+
+namespace App\Services;
+
+use App\Entity\Device;
+use App\Entity\StorageNode;
+use Doctrine\ORM\EntityManagerInterface;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\DependencyInjection\ParameterBag\ParameterBagInterface;
+use Symfony\Component\Process\Exception\ProcessFailedException;
+use Symfony\Component\Process\Process;
+
+class CleanupService
+{
+
+    private $path;
+
+    private $em;
+    private $logger;
+
+    private $inactiveDevices;
+    private $activeDeviceIds;
+    private $storedDeviceIds;
+    private $storedActiveDevices;
+
+    private $storageNode;
+    private $activeProbes;
+    private $activeGroups;
+
+
+    public function __construct(EntityManagerInterface $em, LoggerInterface $logger, ParameterBagInterface $params)
+    {
+        $this->path = $params->get('rrd_storage_path');
+
+        $this->em = $em;
+        $this->logger = $logger;
+
+    }
+
+    public function setStorageNode(StorageNode $storageNode){
+        $this->storageNode = $storageNode;
+    }
+
+    public function cleanup(){
+
+        echo 'Sitrep ' . PHP_EOL;
+
+        $this->logger->info('Retrieving statistics...');
+        $this->getCurrentSituation();
+
+        echo 'Remove Inactive Devices ' . PHP_EOL;
+
+        $this->logger->info('Removing inactive devices...');
+        $this->removeInactiveDevices(10);
+
+        echo 'Sitrep ' . PHP_EOL;
+
+        $this->logger->info('Updating statistics...');
+        $this->getCurrentSituation();
+
+        echo 'Gather Active Probes ' . PHP_EOL;
+
+        $this->logger->info('Gathering active probes...');
+        $this->gatherActiveProbes();
+
+        echo 'Remove Inactive Probes ' . PHP_EOL;
+
+        $this->logger->info('Removing inactive probes...');
+        $this->removeInactiveProbes();
+
+        echo 'Gather Active Slave groups ' . PHP_EOL;
+
+        $this->logger->info('Gathering active slave groups...');
+        $this->gatherActiveSlaveGroups();
+
+        echo 'Remove Inactive Slave Groups ' . PHP_EOL;
+
+        $this->logger->info('Removing inactive slave groups...');
+        $this->removeInactiveSlaveGroups();
+    }
+
+    protected function getCurrentSituation()
+    {
+
+        //create an array of existing directories
+        $this->storedDeviceIds = $this->getDirContent($this->path);
+
+        //remove newline
+        array_pop($this->storedDeviceIds);
```

Comment:
Is it always and only the last element that's a newline? Why doesn't `getDirContents($)` already remove it?